web/themes/custom/novel/novel.theme

```diff
@@ -276,3 +277,43 @@ function novel_theme_suggestions_select_alter(array &$suggestions, array $variab
     $suggestions[] = 'select__single';
   }
 }
+
+/**
+ * Preprocesses rows in a Drupal view to handle the display of event instances.
+ *
+ * This function iterates over each row in the view, checking for the presence
+ * of an EventInstance object.
+ * When an EventInstance is found, it checks if it belongs to the same
+ * event series as the previous row.
+ * If so, it marks the current row as 'stacked' and adjusts its view
+ * mode accordingly.
+ * This is useful for grouping events from the same series in the view,
+ * enhancing visual continuity and clarity.
+ *
+ * @param array &$variables
+ *   An array of variables to pass to the theme layer.
+ */
+function novel_preprocess_views_view_unformatted(&$variables): void {
+  $rows = &$variables['rows'];
+
+  foreach ($rows as $index => &$row) {
+    if (isset($row['content']['#eventinstance'])) {
+      /** @var \Drupal\recurring_events\Entity\EventInstance $eventInstance */
+      $eventInstance = $row['content']['#eventinstance'];
+
+      if ($eventInstance instanceof EventInstance) {
```

Comment:
```suggestion
      $eventInstance = $row['content']['#eventinstance'];

      if ($eventInstance instanceof EventInstance) {
```

The type hint should not be necessary/valuable when a subsequent line actually ensures that the value is of the expected type.